src/Home.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react'
+import {Link} from 'react-router-dom'
+import * as BooksAPI from "./BooksAPI"
+
+import Book from "./Book"
+
+class HomePage extends Component {
+  state = {
+    books: []
+  }
+  componentDidMount() {
+    BooksAPI.getAll().then(books => {
+      this.setState({ books })
+    })
+  }
+  changeShelf = (e, book) => {
+    const books = this.state.books
+    const shelf = e.target.value
+    book.shelf = shelf
+    this.setState({
+      books
+    })
+
+    BooksAPI.update(book, shelf).then(() => {
+      this.setState(state => ({
+        books: state.books
+          .filter(b => b.id !== book.id)
+          .concat([book])
+      }))
+    })
+  }
+  render() {
+    const books = this.state.books
+    const currentlyReading = books.filter(
+      book => book.shelf === "currentlyReading"
+    )
+    const wantToRead = books.filter(
+      book => book.shelf === "wantToRead"
+    )
+    const read = books.filter(
+      book => book.shelf === "read"
+    )
+
+    return(
+      <div className="list-books">
+        <div className="list-books-title">
+          <h1>MyReads</h1>
+        </div>
+        <div className="list-books-content">
+          <div>
+            <div className="bookshelf">
```

Comment:
Lines 51-76 have a lot of duplication. Consider creating some sort of loop.

Reply:
I`ve cleaned a bit this HomePage Shelves constructors. I do not see a reason for creating a loop for this part.